src/Integration/Binding/CSharpVBBindingOperation.cs

```diff
@@ -153,7 +153,9 @@ private void AddRuleset()
         /// </summary>
         private void TrySetNonConditionalRuleSet()
         {
-            var hasUserSpecifiedRuleSet = propertyInformationMap.Keys.Any(x => !IsDefaultMicrosoftRuleSet(x.Value as string));
```

Comment:
The bug was that in the `CalculateRuleSetInformation` method I ignore all rulesets that already reference the generated ruleset.
So propertyInformationMap was empty.